### PR TITLE
Set up GitHub actions to replace Travis

### LINKED ISSUE #190: Update CI workflows
@nickjj suggested we migrate from Travis to GitHub actions... We should definitely do that:
- [ ] Add new GitHub workflows
- [ ] Update the CI links in the readme to point at them instead of Travis: https://github.com/pallets-eco/flask-debugtoolbar/blob/master/README.rst
- [ ] Delete old travis workflows
- [x] I looked into deleting the travis org  for https://travis-ci.org/github/flask-debugtoolbar (xref #188) but it looks like this isn't even possible... the page has a banner at the top about migrating to travis.ci and says builds stopped two years ago.

### PR DESCRIPTION
**Pull request by greyli**
fixes #190 

* Enable GitHub Actions
* Test against Python 2.7, 3.6~3.11
* Update the build badge
* Remove Travis config

The tests can be triggered normally:

https://github.com/greyli/flask-debugtoolbar/pull/1

image

### CODE REVIEW

**Comment by greyli** [comment]
@jeffwidman @nickjj Please help to review & merge this PR to enable the CI. After the CI is enabled, we could merge the following PR to release 0.14 (the CI will be passed after the first two PRs are merged):

- #216
- #211
- #200 (?)

After that, we could work on the 1.0 version and drop support to Python 2.7 and 3.6.

File: .github/workflows/tests.yml
Comment (jeffwidman) [inline .github/workflows/tests.yml:11]: Would we not want to have tests fire on _all_ pull requests? Ie, if someone does a stacked PR, be nice to have tests fire.

File: .github/workflows/tests.yml
Comment (jeffwidman) [inline .github/workflows/tests.yml:38]: Curious why you ran this as two separate commands rather than `pip install -U wheel setuptools`?

To be clear, this isn't a "please change" request, but simply curious to learn.

File: .github/workflows/tests.yml
Comment (jeffwidman) [inline .github/workflows/tests.yml:23]: Was it a conscious decision to have `3.11` as the latest version rather than `3.12`?

**Review by jeffwidman — APPROVED** [review]
This all looks fine to me. A couple of small questions, but I'm going to merge as-is to unblock the other PR's for upgrades, and let @greyli decide on handling my comments on this PR via a follow-up PR.

**Comment by jeffwidman** [comment]
LOL, looks like you went ahead with merge just as I left my review, 👍 from me, just please take a look at some point.

**Comment by nickjj** [comment]
Should we revert and modify this PR to pull in Jeff's suggestions or roll forward? At the very least I do think we'll want the test suite to run on every PR.

**Comment by jeffwidman** [comment]
IMO, we should roll forward. I was planning to merge this myself til @greyli beat me to it. Because having working CI, even if only on `master` is much better than totally broken Travis. 😀

**Comment by greyli** [comment]
Thanks for the review! I think I enabled the auto-merge option... Anyway, I fixed the issues you pointed out and made some more improvements in #217.

**Pull request merged**
